lib/db_connection/connection.ex

```diff
@@ -1,5 +1,17 @@
 defmodule DBConnection.ConnectionError do
-  defexception [:message, severity: :error]
+  defexception [:message, severity: :error, reason: :error]
+
+  @moduledoc """
+  The raised exception might include the reason which would be useful
```

Comment:
I'm not sure if this is the best way to document exceptions. Please feel free to suggest any better way.